module_utils/kafka_manager.py

```diff
@@ -417,10 +445,7 @@ def is_topic_configuration_need_update(self, topic_name, topic_conf):
         configuration entries for a topic, we need to use Zookeeper.
         Requires zk connection.
         """
-        current_config, _zk_stats = self.zk_client.get(
-            self.ZK_TOPIC_CONFIGURATION_NODE + topic_name
-        )
-        current_config = json.loads(current_config)['config']
+        current_config = self.get_config_for_topic(topic_name, None)
```

Comment:
I think it breaks the current behavior of the lib in the given situation:
1. Create topic A with options `{flush.ms: 12345, retention.ms: 6789}`
2. Update topic A with options `{retention.ms: 6789}` (deleting `flush.ms` from the specified options)

Current behavior = `flush.ms` is reset to the default value
New behavior = `flush.ms` will keep its previous value
If I remember well, this is the reason why I actually used Zookeeper here: it only stores options that differs from the default ones. I found this behavior more user friendly, but it would be great to have your thoughts on it.

Reply:
Mh not even sure this is true. Will double check...

Reply:
Turns out I was wrong about the "new behavior", it is currently still the same (I added a test to check it). But I think it is the same because updating the options of a topic is not idempotent anymore since the [next check](https://github.com/StephenSorriaux/ansible-kafka-admin/pull/79/files#diff-4a6d50fd984b69b6a324d60d8267abd633c6cef5d961c06552505f07c78e6fc2R450) is comparing the full options configuration of the topic vs the options configuration given as parameter to the lib

Reply:
I added some modification to make it idempotent.

Reply:
Awesome!